pkg/services/postgresqldb/provision_dbms_only.go

```diff
@@ -0,0 +1,160 @@
+package postgresqldb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/open-service-broker-azure/pkg/generate"
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+	uuid "github.com/satori/go.uuid"
+)
+
+func (d *dbmsOnlyManager) ValidateProvisioningParameters(
+	provisioningParameters service.ProvisioningParameters,
+) error {
+	pp, ok := provisioningParameters.(*ServerProvisioningParameters)
+	if !ok {
+		return errors.New(
+			"error casting provisioningParameters as " +
+				"*postgresql.ServerProvisioningParameters",
+		)
+	}
+	return validateServerParameters(
+		pp.SSLEnforcement,
+		pp.FirewallIPStart,
+		pp.FirewallIPEnd,
+	)
+}
+
+func (d *dbmsOnlyManager) GetProvisioner(
+	service.Plan,
+) (service.Provisioner, error) {
+	return service.NewProvisioner(
+		service.NewProvisioningStep("preProvision", d.preProvision),
+		service.NewProvisioningStep("deployARMTemplate", d.deployARMTemplate),
+	)
+}
+
+func (d *dbmsOnlyManager) preProvision(
```

Comment:
It might not work out, but is it possible to DRY this up? There's a lot in common with the all-in-one pre-provisioning here. The thing that might be problematic is that the type of object that the values are being assigned to varies between the two managers, but _maybe_ we can still cope with that if the all-in-one details are composed of dbms details?? Maybe? i.e. Can we extract most of this logic and _safely_ do a type assertion that a all-in-one-details are db-only details?

Reply:
If you think it makes things harder to grok, I'm also ok with leaving it be in the name of not overcomplicating things. Dealer's choice here.

Reply:
yeah, let me see if that's doable!